ejb3/src/main/java/org/jboss/as/ejb3/timerservice/TimerTask.java

```diff
@@ -122,6 +122,19 @@ public void run() {
                     // execution until it is complete. See JIRA AS7-3119
                     if (timer.getState() == TimerState.IN_TIMEOUT || timer.getState() == TimerState.RETRY_TIMEOUT) {
                         EJB3_TIMER_LOGGER.skipOverlappingInvokeTimeout(timer, now);
+
+                        if (EJB3_TIMER_LOGGER.isDebugEnabled()) {
+                            // WFLY-10542 log thread stack trace which is processing timer task in debug level to diagnose timer overlap
+                            Thread otherThread = timer.getExecutingThread();
```

Comment:
This may be null for clustered timers if the timer is executing on another node.

Reply:
I have updated it to only log debug message for local timer (otherThread is not null)

Reply:
Yes. the executingThread in Builder here is unnecessary. Timer should always be constructing executingThread value to null.